### ISSUE TITLE
Alphabetically sorting with co-located hooks for `scripts`

### ISSUE THREAD

**Issue by rakleed** [matzkoh/prettier-plugin-packagejson#205]
npm `run-scripts` support [`pre` and `post` hooks](https://docs.npmjs.com/cli/v10/using-npm/scripts#pre--post-scripts). For readability I prefer them to be co-located next to their script they belong to,

e.g.

```json
{
  "scripts": {
    "precompress": "",
    "compress": "",
    "postcompress": ""
  }
}
```

while `pre` comes before and `post` after the actual script.

So this should be valid:

```json
{
  "scripts": {
    "build": "",
    "precompress": "",
    "compress": "",
    "postcompress": ""
  }
}
```

and this one is invalid:

```json

{
  "scripts": {
    "build": "",
    "compress": "",
    "postcompress": "",
    "precompress": ""
  }
}
```

---

Same as https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/499

**Comment by matzkoh** [comment]
@rakleed 

This plug-in uses https://github.com/keithamus/sort-package-json internally and does not have its own logic.
Could you please try to find the discussion in the above repository?
I assume you are probably using npm-run-all.

**Comment by matzkoh** [comment]
@rakleed 

No, this plugin merely uses sort-package-json transparently and does not have any custom logic of its own.
If sections other than scripts are being sorted correctly, then this plugin is functioning as intended.
In that case, please report the issue to sort-package-json.

**Comment by rakleed** [comment]
Probably a few weeks ago, for some reason, `pre` and `post` were not taken into account. And now I checked it again, and it works correctly 🤷‍♂️
